go/vt/vtorc/inst/analysis_dao.go

```diff
@@ -590,13 +597,64 @@ func GetReplicationAnalysis(keyspace string, shard string, hints *ReplicationAna
 		return nil
 	})
 
+	for _, analysis := range result {
+		log.Errorf("Analysis - Instance - %v, Code - %v, LastCheckValid - %v, ReplStopped - %v", analysis.AnalyzedInstanceAlias, analysis.Analysis, analysis.LastCheckValid, analysis.ReplicationStopped)
+	}
+
+	result = postProcessAnalyses(result, clusters)
+
 	if err != nil {
 		log.Error(err)
 	}
 	// TODO: result, err = getConcensusReplicationAnalysis(result)
 	return result, err
 }
 
+// postProcessAnalyses is used to update different analyses based on the information gleaned from looking at all the analyses together instead of individual data.
+func postProcessAnalyses(result []*ReplicationAnalysis, clusters map[string]*clusterAnalysis) []*ReplicationAnalysis {
```

Comment:
I feel like this function is not needed? There's the "mega query" above that already joins a primary and its replicas. For example, detecting a `DeadPrimary` is done by detecting that the primary is unreachable _and_ all of its replicas are unreachable.
I feel like it should be possible to use that same query to see if all replicas agree that the primary is dead.

Reply:
I actually started with trying it out that way. But consider this situation, VTOrc starts up after a vttablet has crashed. In this case, we only have a vttablet record of the primary, and that record too has an empty hostname and port. So, there is no way to correlate this primary with the other replicas. In the join we do in the mega query, we actually see no replicas for this tablet (empty hostname, port, so the join doesn't find any replicas that are replicating from `:0`).
Therefore, I have to do a post-process phase, where I check the replicas from the same keyspace-shard and check if they are able to replicate or not.
If none of the replicas can replicate, then I conclude we can change the `InvalidPrimary` to a `DeadPrimary` recovery.

Reply:
Thank you for clarifying. I see the problem now, sounds good.